src/sgame/sg_bot.cpp

```diff
@@ -445,12 +445,29 @@ void G_BotThink( gentity_t *self )
 		//BotClampPos( self );
 	}
 
+	self->botMind->willSprint( false ); //let the BT decide that
 	self->botMind->behaviorTree->run( self, ( AIGenericNode_t * ) self->botMind->behaviorTree );
 
 	// if we were nudged...
 	VectorAdd( self->client->ps.velocity, nudge, self->client->ps.velocity );
 
+	// ensure we really want to sprint or not
 	self->client->pers.cmd = self->botMind->cmdBuffer;
+	int const* stats = self->client->ps.stats;
+	int jumpCost = BG_Class( stats[ STAT_CLASS ] )->staminaJumpCost;
+	int stamina = stats[ STAT_STAMINA ];
+
+	self->botMind->m_exhausted = self->botMind->m_exhausted || ( self->botMind->botSkill.level >= 5 && stamina <= jumpCost + jumpCost / 10 );
```

Comment:
I find this more readable (assuming you want to keep the current logic):
```suggestion
	if ( self->botMind->botSkill.level >= 5 && stamina <= jumpCost + jumpCost / 10 )
	{
		self->botMind->m_exhausted = true;
	}
```

But I'm wondering if you really need to set the variable twice: the current logic seems equivalent to this:
```C++
	self->botMind->m_exhausted = stamina <= jumpCost * 2 || ( self->botMind->botSkill.level >= 5 && stamina <= jumpCost + jumpCost / 10 );
```

Which is in turn equivalent to this:
```suggestion
	self->botMind->m_exhausted = stamina <= jumpCost * 2 || ( self->botMind->botSkill.level >= 5 && stamina <= jumpCost * 1.1f );
```

I think instead of jumpCost, you intended to use the skill, but that, assuming I did read right until now, would be equivalent to:
```suggestion
	self->botMind->m_exhausted = stamina <= jumpCost * 2;
```

So maybe we can just go with that for now?

Reply:
Have you tried your proposed changes?